tyxml/snabbdom_tyxml.ml

```diff
@@ -65,10 +65,14 @@ end
 
 module Svg_vnode = struct
   include Xml_vnode
+end
+
+module Svg = struct
+  include Svg_f.Make (Svg_vnode)
 
-  let node ?a name children =
-    Vnode.make_node "svg" Jv.null [ Vnode.make_node name (make_attrs a) children ]
+  (* overriding the [svg] function generated by the tyxml functor since it adds some
+     namespace attribures. We don't need those when creating svg nodes using snabbdom. *)
+  let svg ?a children = Vnode.make_node "svg" (Xml_vnode.make_attrs a) children
 end
 
-module Svg = Svg_f.Make (Svg_vnode)
 module Html = Html_f.Make (Xml_vnode) (Svg)
```

Comment:
You forgot the `.mli`/signatures for `Html` and `Svg`. The signatures are essential to make the tyxml typing works. See https://ocsigen.org/tyxml/latest/manual/functors#sig for the explanations.

Reply:
So i added the mli module, but I don't see an error when using the `Svg.svg` example i had before. I also tried it with regular tyxml under utop and tyxml seems to accept it without issues. I'm also not sure I'm doing the right thing here with the signatures. The docs recommend not exposing the `+'a elt = Xml.elt` equality, but without that I won't be able to forward the tyxml output to the snabbdom patch function (the core library expects to work with vnodes)

Reply:
You should most definitely *not* expose that equality! It bypasses tyxml's typing.

When you want to obtain some `Xml.elt` (in your case, to render it), you need to use `Html/Svg.toelt`. I suggest you expose a function that renders an `'a elt` directly.

Reply:
> You should most definitely **not** expose that equality! It bypasses tyxml's typing.

The newest commits should ensure I don't expose these in the mli. I'm working via `toelt` at the moment. I think eventually i'll try to have everything needed to use snabbdom re-exported in the tyxml specific package too so it exposes its own init/patch functions that work with tyxml elements and the user can stick to just working with a `Snabbdom_tyxml` module.